### ISSUE TITLE
[v2020.08.07] Back Row didn't reduce damage 

### ISSUE THREAD

**Issue by DV666** [Albeoris/Memoria#148]
**Memoria version :** v2020.08.07 (last)

Hello everyone,

First of all, thank you to the team for updating Memoria quickly after the FFIX update : thanks all ! :D

So I was able to update my mod and I think I noticed a little problem with the new version of Memoria : indeed, the damage on the second raw is the same as on the first raw.

It seems that the following code in the CSharp poses a problem (Memoria folder) :

```
public void BonusBackstabAndPenaltyLongDistance()
{
	bool flag = this.IsCasterSameDirectionTarget() || this.Target.IsRunningAway();
	if (flag)
	{
		this.Context.Attack = this.Context.Attack * 3 >> 1;
	}
	bool flag2 = Mathf.Abs((int)(this.Caster.Row - this.Target.Row)) > 1 && !this.Caster.HasLongRangeWeapon && this.Command.IsShortRange;
	if (flag2)
	{
		this.Context.Attack /= 2;
	}
}

```
Modifying this part seem to fix the problem:

```
public void BonusBackstabAndPenaltyLongDistance()
{
	if (this.IsCasterSameDirectionTarget() || this.Target.IsRunningAway())
	{
		this.Context.Attack = this.Context.Attack * 3 >> 1;
	}
	if (Mathf.Abs((int)(this.Caster.Row - this.Target.Row)) > 1 && !this.Caster.HasLongRangeWeapon)
	{
		this.Context.Attack /= 2;
	}
}

```

If you can check sideways if I'm not saying anything stupid to fix it for the next update (maybe ASAP since it's a pretty major problem, no ? ^^)

Good day to you and thanks again ^^

**Comment by snouz** [comment]
Hi, so I'm checking the code:

public void BonusBackstabAndPenaltyLongDistance()
        {
            if (IsCasterSameDirectionTarget() || Target.IsRunningAway())
                Context.Attack = Context.Attack * 3 >> 1;

            // Note that there are two weapon categories: SHORT_RANGE and LONG_RANGE
            // LONG_RANGE is used for this penalty while SHORT_RANGE is used both for that and for "out of range" enemies
            if (Mathf.Abs(Caster.Row - Target.Row) > 1 && !Caster.HasLongRangeWeapon && Command.IsShortRange)
                Context.Attack /= 2;
        }

What you say is that Command.IsShortRange always returns false, right?

I'll ping Tirlititi, I think he added that condition.

**Comment by Albeoris** [comment]
@Tirlititi , check this commit please:
https://github.com/Albeoris/Memoria/commit/bfb2fd3060252a977a9554ea8aa8a98aa7d22e37

**Comment by DV666** [comment]
> 
> What you say is that Command.IsShortRange always returns false, right?
> 
> I'll ping Tirlititi, I think he added that condition.

There's a good chance. I'm not at all a C# pro like you, but since the last update, the second raw has a problem.

The correction I made on this function seems to correct the problem for my part.

**Comment by Tirlititi** [comment]
Should be fully fixed with commit 3810875 (both with CustomBattleFlagsMeaning = 0 and CustomBattleFlagsMeaning = 1).
